be/src/exprs/string_functions.cpp

```diff
@@ -99,6 +99,32 @@ BooleanVal StringFunctions::ends_with(
     return BooleanVal(str_sp.ends_with(suffix_sp));
 }
 
+BooleanVal StringFunctions::empty(
+        FunctionContext* context, const StringVal& str) {
+    if (str.is_null) {
+        return BooleanVal::null();
```

Comment:
Considered non-empty if it contains at least one byte, even if this is a space or a null byte. The function should return true or false, not null.